src/components/inputs/Search/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactComponent as SearchIcon } from '../../../icons/Search.svg';
+import { ReactComponent as ArrowUpRight } from '../../../icons/ArrowUpRight.svg';
+import { useState, useEffect } from 'react';
+import './styles.css';
+import { SearchBarProps } from '../../../../types';
+
+function SearchBar(props: SearchBarProps) {
+  const {
+    buttonAction,
+    searchBarState,
```

Comment:
this requires parent components to have context for various states of the search bar, which we don't necessarily want. instead, we can manage states internally so we don't have to reimplement what they are in parent components everytime this search bar is used.

TLDR, remove `searchBarState` and `onChangeHandler` from props, and we can add internal functions that mange those later on